File: device-app/android/Java/umsDemo_prev/app/src/main/java/com/rationalowl/umsdemo/domain/UserRepository.java
package com.rationalowl.umsdemo.domain;

public class UserRepository extends JsonRepository<User> {
    private static final String FILE_NAME = "user.json";

    private static UserRepository instance;

    public static synchronized UserRepository getInstance() {
        if (instance == null) {
            instance = new UserRepository();
        }

        return instance;
    }

    private UserRepository() {
        super(User.class, FILE_NAME);
    }

    public User getUser() {
        return getValue();
    }

    public void setUser(User user) {
        setValue(user);
    }
}

File: device-app/ios-swift/umsDemo_prev/ums/Domain/UserRepository.swift
class UserRepository : JsonRepository<User>
{
    private static let fileName = "user.json"
    
    static let shared = UserRepository()
    
    private init()
    {
        super.init(UserRepository.fileName);
    }
    
    var user: User?
    {
        get
        {
            return getValue()
        }
        set
        {
            setValue(newValue!)
        }
    }
}
